### PR TITLE
RFD 322: v1 built-in users

### PR DESCRIPTION
**Pull request by david-crespo**
Roles don't have IDs exposed in the responses so it doesn't make sense to make them fetchable by ID.

### CODE REVIEW

File: nexus/src/external_api/http_entrypoints.rs [hunk 1]
```diff
@@ -8890,6 +8953,75 @@ async fn role_view(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// List built-in roles
+#[endpoint {
+    method = GET,
+    path = "/v1/roles",
```
Comment (zephraph) [inline nexus/src/external_api/http_entrypoints.rs:8959]: So this was a failure on my part when introducing `/system`. This resource is fleet scoped and should be moved under `/system`. Also, given that both `roles` and `users` apply to builtin we could do `/v1/built-in/users` and `/v1/built-in/roles`. I'd also like to find a term better than `built-in`, but that can be deferred.

Comment (zephraph) [reply nexus/src/external_api/http_entrypoints.rs:8959]: See the fleet scoping of roles here: https://github.com/oxidecomputer/omicron/blob/369af0ee7ab0bf70dc47422126804b19117dd850/nexus/db-queries/src/db/datastore/role.rs#L43

Comment (david-crespo) [reply nexus/src/external_api/http_entrypoints.rs:8959]: Definitely on `/system`, that makes perfect sense. Not sure about `/built-in/...`. It is clean, but a built-in is not really a thing, so maybe it's misleading?

Comment (david-crespo) [reply nexus/src/external_api/http_entrypoints.rs]: Yeah, thinking more about it, I think the fact that they're built in is an implementation detail I'd rather not expose through the API for now. We can revisit that later when we have roles that are not built in. The built in users thing is different. I'm not sure we even need these endpoints.

Comment (zephraph) [reply nexus/src/external_api/http_entrypoints.rs]: > the fact that the roles are built in is an implementation detail I'd rather not expose through the API for now

That was my initial inclination for both. Users (aside from "built-in" users) don't exist at a system context so `/system/users` made sense to me. I think the same can be said of roles. They're both denoting oxide provided resources.

Comment (david-crespo) [reply nexus/src/external_api/http_entrypoints.rs]: Problem is we are using `/system/users?silo={silo}`.

**Pull request merged**